deploy/dynamo/sdk/src/dynamo/sdk/cli/env.py

```diff
@@ -0,0 +1,174 @@
+    # TODO: Revisit once we need to support Windows based systems
+    return f"{distro.name()} {distro.version()}"
+
+
+def execute_subprocess_output(command: str) -> str:
+    """Execute a subprocess command and return the output."""
+    try:
+        out = subprocess.check_output(command, shell=True, stderr=subprocess.DEVNULL)
+        if not out.strip():
+            return "N/A"
+        return out.decode("utf-8").strip()
+    except subprocess.CalledProcessError:
+        return "N/A"
+
+
+def get_glibc_version() -> str:
+    """Get GLIBC version."""
+    return execute_subprocess_output("ldd --version | head -n 1 | awk '{print $NF}'")
+
+
+def get_gcc_version() -> str:
+    """Get GCC version."""
+    return execute_subprocess_output("gcc --version | head -n 1 | awk '{print $NF}'")
+
+
+def get_cmake_version() -> str:
+    """Get Cmake version."""
+    return execute_subprocess_output("cmake --version | head -n 1 | awk '{print $NF}'")
+
+
+def get_rust_version() -> str:
+    """Get Rust version."""
+    return execute_subprocess_output(
+        "rustc --version | head -n 1 | awk '{print $(NF-2)}'"
+    )
+
+
+def get_docker_version() -> str:
+    """Get Docker version."""
+    return execute_subprocess_output("docker --version | awk '{print $3}' | tr -d ','")
+
+
+def get_cpu_architecture() -> str:
+    """Get CPU architecture."""
+    return execute_subprocess_output("lscpu")
+
+
+def query_nvidia_smi(param: str) -> str:
+    """Get GPU information from nvidia-smi if available"""
+    return execute_subprocess_output(
+        f"nvidia-smi --query-gpu={param} --format=csv,noheader"
+    )
+
+
+def get_gpu_topo() -> str:
+    """Get GPU topology if available"""
+    return execute_subprocess_output("nvidia-smi topo -m")
+
+
+def get_cuda_version() -> str:
+    """Get CUDA version if available."""
+    return execute_subprocess_output(r"nvcc --version | grep -Po 'release \K\d+\.\d+'")
+
+
+def get_python_platform():
+    return platform.platform()
+
+
+def get_installed_packages() -> list[tuple[str, str]]:
+    """Get list of installed Python packages and their versions."""
+    return [(pkg.key, pkg.version) for pkg in pkg_resources.working_set]
+
+
+def get_python_packages() -> str:
+    """Get list of specified Python packages and their versions."""
+    installed_packages = get_installed_packages()
+    out = []
+    search_python_packages = [
+        "ai-dynamo",
+        "ai-dynamo-runtime",
+        "ai-dynamo-vllm",
+        "genai-perf",
+        "nixl",
+        "numpy",
+        "nvidia-cublas-cu12",
+        "nvidia-cuda-cupti-cu12",
+        "nvidia-cuda-nvrtc-cu12",
+        "nvidia-cuda-runtime-cu12",
+        "nvidia-cudnn-cu12",
+        "nvidia-cufft-cu12",
+        "nvidia-curand-cu12",
+        "nvidia-cusolver-cu12",
+        "nvidia-cusparse-cu12",
+        "nvidia-ml-py",
+        "nvidia-nccl-cu12",
+        "nvidia-nvjitlink-cu12",
+        "nvidia-nvtx-cu12",
+        "pyzmq",
+        "tensorrt_llm",
```

Comment:
`vllm` and `sglang` too?

Reply:
Oh I see `ai-dynamo-vllm` is there instead - maybe we'll swap to `vllm` when we upstream our patch changes.